internal/api/request.go

```diff
@@ -98,15 +99,33 @@ func (apiResponse *Response) GetHeader(name string) string {
 
 func retryAfter(ctx context.Context, resp *http.Response) time.Duration {
 	retryHeader := resp.Header.Get(constants.HEADER_RETRY_AFTER)
+	if retryHeader == "" {
+		return DefaultRetryAfter()
+	}
 	tflog.Debug(ctx, "Retry Header: "+retryHeader)
 
-	retryAfter, err := time.ParseDuration(retryHeader)
-	if err != nil {
-		// default retry after 5-10 seconds
-		return DefaultRetryAfter()
+	// Check if the header is a delta-seconds value (integer)
+	if deltaSeconds, err := strconv.Atoi(retryHeader); err == nil {
+		return time.Duration(deltaSeconds) * time.Second
+	}
+
+	// Check if the header is an HTTP-date
+	if retryTime, err := http.ParseTime(retryHeader); err == nil {
+		// Calculate duration until the retry time
+		duration := time.Until(retryTime)
+		if duration > 0 {
+			return duration
+		}
+	}
+
+	// Try to parse as a duration string (non-standard but sometimes used)
+	if retryAfter, err := time.ParseDuration(retryHeader); err == nil {
+		return retryAfter
 	}
 
-	return retryAfter
+	// Fallback to a default retry duration
+	tflog.Warn(ctx, "Invalid Retry-After header, falling back to default")
```

Comment:
we'll have to see if we want this to be a warn or just debug.  could flood the warnings log with something the user can't do anything about.